greenriver/hmis-warehouse
# Accept signed blob ID when uploading images

## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
GH issue: https://github.com/open-path/Green-River/issues/7077 ( https://github.com/open-path/Green-River/issues/6208 )
Corresponding frontend PR: https://github.com/greenriver/hmis-frontend/pull/987

This PR updates the (for now, two) places in HMIS where we accept a blob ID: the client headshot uploader, and the file processor. These now accept a signed blob ID, perhaps partially addressing the concern Gig pointed out here https://github.com/open-path/Green-River/issues/6968#issuecomment-2514886196.

For backwards compat, these endpoints also still accept an unsigned blob ID. That should be turned off on the next release.

I split this work out of the other PR since I thought it would be helpful to be able to discuss independently.

QA instructions
- I should be able to upload a client headshot
- I should be able to re-upload a new client headshot for a client that already has one
- I should be able to submit a Client File

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [ ] I have updated the documentation (or not applicable) - documentation to be done in https://github.com/open-path/Green-River/issues/7054

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected
